From cd8fc705e7024da120a4e9f61c5c85b5d4b4818a Mon Sep 17 00:00:00 2001
From: "T R, Thejesh Reddy" <thejesh.reddy.t.r@intel.com>
Date: Fri, 8 Oct 2021 11:06:35 +0530
Subject: [PATCH 055/109] stmmac fix for HSD1509042892

Clearing mss in TDES to indicate it is new TSO transmission when it is
enabled from TSO off using ethtool

Signed-off-by: T R, Thejesh Reddy <thejesh.reddy.t.r@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index ff7e9812c1b6..0c86362a28dc 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -5714,11 +5714,8 @@ static netdev_features_t stmmac_fix_features(struct net_device *dev,
 
 	/* Disable tso if asked by ethtool */
 	if ((priv->plat->tso_en) && (priv->dma_cap.tsoen)) {
-		if (features & NETIF_F_TSO) {
-			struct stmmac_tx_queue *tx_q = &priv->tx_queue[0];
+		if (features & NETIF_F_TSO)
 			priv->tso = true;
-			tx_q->mss = 0;
-		}
 		else
 			priv->tso = false;
 	}
@@ -5733,6 +5730,7 @@ static netdev_features_t stmmac_fix_features(struct net_device *dev,
 		tx_q->mss = 0;
 		stmmac_enable_tso(priv, priv->ioaddr, priv->tso, chan);
 	}
+
 	return features;
 }
 
-- 
2.25.1

